Fja 73 74 75 api new fields

alireid opened the pull request.

alireid committed:
Initial commit

Changes to facilitate new fields and returned based on new dates in table

alireid committed:
Tweaks

alireid committed:
Using DateTimeProvider

alireid committed:
Update tests

alireid committed:
Update GetAgencyQueryTests.cs

alireid committed:
Tests update

alireid committed:
Add null checking for end date

alireid committed:
WIP

alireid committed:
WIP

alireid committed:
Update IAgencyRepository.cs

alireid committed:
Example scripts updated

alireid committed:
A necessary file for build it seems

alireid committed:
Tests updates

alireid committed:
Clean up post tech review

alireid committed:
Minor tweaks for sonarcloud

alireid committed:
Changed so no ID field needed

Using LegalEntityId and EffectiveFrom as compound primary

alireid committed:
Order by CreatedDate descending and select the first item

alireid committed:
Updates to not include elements from 74

This branch does not include FJA-74, 74 includes features such as EffectiveTo dates used in LINQ queries and RemovalReason. Features of this branch will only include agencies to be shown EffectiveFrom date > greater than Now.

alireid committed:
Removal table fields

alireid committed:
Removed tests as cant use to and from given removal of FJA74

alireid committed:
Delete SFA.DAS.Rofjaa 2.sln

chrisfoster76 commented on src/SFA.DAS.Rofjaa.Database/Scripts/AgencyImportTest.sql:
```diff
@@ -1,2 +1,2 @@
-﻿INSERT INTO [dbo].[Agency] VALUES (1, 1);
-INSERT INTO [dbo].[Agency] VALUES (2, 0);
\ No newline at end of file
+﻿INSERT INTO [dbo].[Agency] (LegalEntityId, IsGrantFunded, EffectiveFrom, EffectiveTo) VALUES (1, 1, '2020/01/01', '2025/01/01');
```
Technically, this script won't work without the EffectiveTo field, I expect

alireid replied:
Good spot! Removed and committed

alireid committed:
Update AgencyImportTest.sql

alireid committed:
Update table allow nulls EffectiveFrom

alireid committed:
EffectiveFrom not nullable

alireid committed:
Reintroduce FJA74

As this ticket status has changed, to now include bringing it back into this branch
